### ISSUE TITLE
Attribute casting seem to work only in one direction

### ISSUE THREAD

**Issue by lukasmu** [DirectoryTree/LdapRecord#707]
It is possible to define the `$casts` property on models as explained here: https://ldaprecord.com/docs/core/v3/model-mutators/#attribute-casting

However, the casting operation seems to work only when getting attributes, not when setting attributes. This is different from how casting works in Laravel's Eloquent and thus confused me.

Not sure if it's easily possible to add this, otherwise I would recommend to update the documentation.

Example:
```
namespace App\Models\Ldap;

use LdapRecord\Models\ActiveDirectory\User as BaseUser;

class User extends BaseUser
{
    protected array $casts = [
        'msExchHideFromAddressList' => 'boolean',
    ];
}
```

What works:
```
$user = User::find('cn=John Doe,dc=local,dc=com');
assert(is_bool($user->msExchHideFromAddressList));
````

What doesn't work:
```
$user = User::find('cn=John Doe,dc=local,dc=com');
$user->msExchHideFromAddressList = false;
$user->save();
````

**Comment by stevebauman** [comment]
Thanks for the report @lukasmu! Apologies for the delayed response on this issue. I'd classify this as a bug -- it should definitely occur both ways. I think this is the case with some casts and not others. Let me update this and make sure its consistent across the board.

**Comment by stevebauman** [comment]
This has been now resolved in [v3.6.2](https://github.com/DirectoryTree/LdapRecord/releases/tag/v3.6.2). Run `composer update` and you're all set. Thanks!
